aiocache/base.py

```diff
@@ -87,6 +91,7 @@ async def _plugins(self, *args, **kwargs):
         return _plugins
 
 
+# class BaseCache(Generic[CacheKeyType]):
```

Comment:
Here and elsewhere in comments are how we might implement typing of the `build_key()` return type if we decide using Generics would be more useful or robust.

Reply:
It probably means also updating tests that check the call signature of `__init()__`, and maybe a few others.

Reply:
I don't think anything needs to be changed to `__init__()`.

Reply:
I was referring to this potential change [_below_], which was intended as a convenience...providing the default`build_key()` return type of `str` so that it could be inferred by mypy, even when not explicitly created with `BaseCache[str]`.

https://github.com/aio-libs/aiocache/blob/84ce34c07b1e2961583c9f3beeb2c146da2ba3d4/aiocache/base.py#L118-L126
[_although the last line should have been `key_type:Type[CacheKeyType] = str`_]

But, unfortunately, it's not (yet) possible to supply a [default type for the `TypeVar`](https://github.com/python/mypy/issues/4236). Oh well.

Reply:
You can use overloads to achieve that, but I don't think it's relevant. There should be no parameter here, all subclasses should explicitly define the type they use.

Reply:
The extra `__init__()` parameter for implicit typing idea has been scrapped.